src/components/AddTaskForm.tsx

```diff
@@ -0,0 +1,73 @@
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { taskSchema, type TaskFormData } from "../schemas/taskSchema";
+
+type TaskFormProps = {
+  onSubmit: (data: TaskFormData) => void;
+};
+export const TaskForm: React.FC<TaskFormProps> = ({ onSubmit }) => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm({
+    resolver: zodResolver(taskSchema),
+    mode: "onChange",
+    defaultValues: {
+      completed: false,
+    },
+  });
+  return (
+    <form
+      onSubmit={handleSubmit((data) => {
+        onSubmit(data);
+        reset();
+      })}
+      className="form"
+    >
+      <div className="form-group">
+        <label>Task Name</label>
+        <input {...register("name")} placeholder="Enter a task name" />
+
+        {errors.name && <p className="error">{errors.name.message}</p>}
+      </div>
+      <div className="form-group">
+        <label>Priority</label>
+        <select {...register("priority")}>
+          <option value="">Select priority</option>
+          <option value="Urgent">Urgent</option>
+          <option value="High">High</option>
+          <option value="Normal">Normal</option>
+          <option value="Low">Low</option>
```

Comment:
We don't have a lot of options here, so this is okay, but personally I'd prefer to handle this with an array, this way if we ever want to change anything, we don't have to touch the code, only the object/array:
```

{priorities.map(value=>
<option value={value.id}>{value.title}</option>
```